### PR TITLE
Fix/scroll lock

### PR DESCRIPTION
**Pull request by Crash--**
- If the content of the Dialog was > on the view port size, then the scroll was not working anymore after opening a dialog from an ActionMenu (because of a conflict between MUI and body-scroll-lock) 

- Disable scroll when BottomDrawer is displayed 

- Disable scroll when Dialog is opened

https://crash--.github.io/cozy-ui/react

Discussion on MUI side: https://github.com/mui-org/material-ui/issues/5750#issuecomment-471806500

### CODE REVIEW

File: react/ActionMenu/index.jsx [hunk 1]
```diff
@@ -56,8 +57,6 @@ const ActionMenuWrapper = ({
     >
       <ClickAwayListener onClickAway={onClose}>{children}</ClickAwayListener>
     </div>
-  ) : (
-    <BottomDrawer onClose={onClose}>{children}</BottomDrawer>
```
Comment (ptbrowne) [inline react/ActionMenu/index.jsx:60]: Why this change ? All the hooks call after the early return are conditional now :/

File: react/Overlay/index.jsx [hunk 2]
```diff
@@ -54,13 +47,14 @@ class Overlay extends Component {
   render() {
     const { children, className } = this.props
     const domProps = omit(this.props, nonDOMProps)
+    const Wrapper = bodyTallerThanWindow() ? React.Fragment : RemoveScroll
```
Comment (ptbrowne) [inline react/Overlay/index.jsx:50]: Why do we have to check this ? Shouldn't we render RemoveScroll in all cases ? If it's necessary, could you add a comment why ?

**Comment by ptbrowne** [comment]
Do you know how react-remove-scroll avoids the issue of body-scroll-lock ?

File: react/ActionMenu/index.jsx (hunk 1 above)
Comment (Crash--) [reply react/ActionMenu/index.jsx:60]: Oh you right, `inline` can change during life cycle. I'll go deeper and extract the react hooks in a new component to fix the point. 

The idea was : 
We don't change viewport size pretty often, and in mobile at least, we don't need to call `usePopper` and all.

File: react/Overlay/index.jsx (hunk 2 above)
Comment (Crash--) [reply react/Overlay/index.jsx:55]: Done and issue created too 👍 https://github.com/cozy/cozy-ui/issues/1599

**Comment by Crash--** [comment]
> Looking good, but I'd like to know more what makes react-remove-scroll work and not body-scroll-lock. Do you know how react-remove-scroll avoids the issue of body-scroll-lock ?

Yep: To be concise, react-remove-scroll is not adding inline style to `body` but instead is create a new inline `<style>` in our HTML document (setting body { overflow: hidden...}). 

By itself, react-remove-scroll is not resolving exactly the issue, but since it uses a different paradigm we don't have the conflict anymore (between MUI and scroll locker) 

I say `not exactly resolving the issue`, but I can be wrong, it seems that `react-remove-scroll` generates an `id` for each `animations` and by doing so it seems to be able to compute the right style at the right moment

**Review by JF-Cozy — APPROVED** [review]

**Comment by ptbrowne** [comment]
Ah ok, it's clever to use a new style element. This way the component only has to add/remove the style element, and it does not have to keep track of the previous value, it's handled by the browser.

**Review by ptbrowne — APPROVED** [review]

**Comment by Crash--** [comment]
argos is broken because I added an exemple with a long text in `Dialog`. I think it's ok to merge ;)

**Comment by JF-Cozy** [comment]
> argos is broken because I added an exemple with a long text in `Dialog`. I think it's ok to merge ;)

Tu devrais pouvoir le mettre à jour pour le faire passer au vert normalement, non ?

**Comment by Crash--** [comment]
Oui, c'est ce que je vais faire, mais je laisse le temps @ptbrowne de remettre son 👍

**Comment by ptbrowne** [comment]
In Argos 

* on the "Dialog-800x600.png" example, the overlay is much darker, is it expected ?
* on the "Dialog-300x600.png" example, the text seems to go above the buttons, is it expected ?

**Review by ptbrowne — APPROVED** [review]
